src/apps/testapps/testCompact.c

```diff
@@ -152,6 +152,52 @@ SUITE(compact) {
                  "compact fails on duplicate input");
     }
 
+    TEST(compact_duplicateMinimum) {
+        // Test that the minimum number of duplicate hexagons causes failure
+        H3Index h3;
+        // Arbitrary index
+        setH3Index(&h3, 5, 0, 2);
+
+        int arrSize = H3_EXPORT(maxH3ToChildrenSize)(h3, 11) + 1;
+        H3Index* children = calloc(arrSize, sizeof(H3Index));
+
+        H3_EXPORT(h3ToChildren)(h3, 11, children);
+        // duplicate one index
+        children[arrSize - 1] = children[0];
+
+        H3Index* output = calloc(arrSize, sizeof(H3Index));
+
+        int compactResult = H3_EXPORT(compact)(children, output, arrSize);
+        t_assert(compactResult != 0,
```

Comment:
Should we assert specific error codes here? Should we have named constants for the error codes?

Reply:
I've updated the PR to have that for the new tests (not updating the existing ones in this PR.) I think this is reasonable when the return codes are named.